src/core/lib/transition-manager.js

```diff
@@ -65,7 +62,7 @@ export default class TransitionManager {
     this.props = nextProps;
 
     if (this._isTransitionEnabled(nextProps)) {
-      const currentViewport = this.state.viewport || extractViewportFrom(currentProps);
+      const currentViewport = this.state.viewport || currentProps;
```

Comment:
Maybe not for this PR, but I don't think we should call this `this.state.viewport`. The interpolator class seems to have generalized the naming `arePropsEqual` etc so why not here.

Details: I don't think the viewport naming is good for two reasons:
1. It would be great if this class develops into a general interpolation manager, could be used for layer attribute and property transitions as well.
2. `viewport` is an overloaded term (that to me currently mainly refers the `Viewport` hierarchy of classes focusing on mathematical projections/unprojections and matrix generations)

* so maybe call this something else like `interpolatedProps` or `interpolatedState` (I can't tell if it is the initial or current props) 
* or maybe just `viewState` if it needs to be view specific).